Odata-docs/webapi-8/fundamentals/server-driven-paging.md

```diff
@@ -0,0 +1,251 @@
+            "Name": "Product 3"
+        },
+        {
+            "Id": 4,
+            "Name": "Product 4"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/Products?$skip=4"
+}
+```
+
+And finally, the last page:
+```http
+GET http://localhost:5000/Products?$skip=4
+```
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 3,
+            "Name": "Product 3"
+        }
+    ]
+}
+```
+
+The last page does not include a next link. This is how the client knows that it has reached the end of the collection.
+
+If the request contains query options, they will also be included in the generated next link. Let's take the following request for example:
+
+```http
+GET http://localhost:5000/Products?$filter=Price gt 1000&$select=Name
+```
+
+The generated next link will look like:
+```
+http://localhost:5000/Products?$filter=Price%20gt%201000&$select=Name&$skip=2
+```
+
+## Combining `PageSize` and `$top`
+
+`PageSize` automatically limits the number of items in the response. The client can still apply `$top` to the request. In this case, `$top` will limit the total number of results rather than the number of items per page.
+
+If the client-requested `$top` is greater than `PageSize`, then the service should return the first page of results, with a next link that fetches the next page of results up to the maximum specified in the client's `$top`. If the `$top` is less that then page size, then server will return the number of items specified in the `$top` query option with no next link.
+
+To illustrate this, let's use our entity set of 5 products as an example, with `PageSize` set to 2. The client makes a request with `$top=3`
+
+```http
+GET http://localhost:5000/Products?$top=3
+```
+**Response:**
+
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Product 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Product 2"
+        }
+    ],
+    "@odata.nextLink": "http://localhost:5000/Products?$top=1&$skip=2"
+}
+```
+The response contains 2 products and a next link to the next page. The client fetches the next page using the next link:
+
+```http
+GET http://localhost:5000/Products?$top=1&$skip=2
+```
+**Response:**
+```json
+{
+    "@odata.context": "http://localhost:5000/$metadata#Products",
+    "value": [
+        {
+            "Id": 3,
+            "Name": "Product 3"
+        }
+    ]
+}
+```
+This is now the last page. The server returns only one item and no next link.
+
+## Improving paging with `$skiptoken`
+
+You may have noticed that ASP.NET Core OData generates next links using `$skip`. Using `$skip` on large collections may lead to performance degradation in some databases. For more information about the drawbaks of `$skip`, visit the [client-driven paging](/odata/webapi-8/fundamentals/client-driven-paging#drawbacks) article.
```

Comment:
I don't understand this paragraph. I expected that since this article talks about server-side paging, it'd focus on enabling skip token feature (`options.SkipToken()`) and applying `PageSize` configuration. My understanding was that those are the two cogs that drive server-side paging. However, what you're saying here is that `$skip` can be used to support server-side paging. That I wasn't aware. I might need to refresh my knowledge. I actually expected this article mostly on `$skiptoken` rather than `$skip`. It's also the reason why I left a lot of review suggestions to "fix" the next link

Reply:
The way I understand it, server-side paging means relying on the server-generated next link. The next link can be generated whether or not you use `$skiptoken`. The next link is computed when you set the `PageSize` property. Skip tokens provide an extra layer of improvements on top of next link. It's also something I discovered while writing this docq when I tried things out and referencing older docs we have on the subject:
- https://learn.microsoft.com/en-us/odata/webapi/pagesize-top-maxtop
- https://learn.microsoft.com/en-us/odata/webapi/skipToken-for-server-side-paging
- https://devblogs.microsoft.com/odata/using-skiptoken-for-paging-in-asp-net-odata-and-asp-net-core-odata/

I could focus only on `$skiptoken`, but I felt it would be incomplete. And given the common issues we get where customers confuse the roles of `PageSize` and `$top`, I thought it would be beneficial to dedicate a section on that. And given that `$skiptoken` is an improvement, I thought it would make more sense to "lead into it". I also considered that someone may come to this page without having read the client-driven paging docs. Open to suggestions.

Reply:
@habbes Our offline conversation refers. It was a knowledge gap on my part. I'm happy with the document covering both `$skip` and `$skiptoken`